src/proxy/mod.rs

```diff
@@ -209,4 +211,55 @@ pub fn to_canonical_ip(ip: SocketAddr) -> IpAddr {
     }
 }
 
+pub fn get_original_src_from_xff(req: &Request<Body>) -> Option<IpAddr> {
+    let value = req.headers().get("X-Forwarded-For")?;
+    let ip = value
+        .to_str()
+        .unwrap_or("")
+        .to_string()
+        .split(',')
+        .next()?
+        .trim()
+        .parse::<IpAddr>();
```

Comment:
maybe you can try map_or() here. for simplicity

Reply:
Thanks for sharing the info. Seems `map_or()` and `ok()` convenient ways to simplify the match blocks.